### ISSUE TITLE
On '2.6 Git Basics - Tagging', branching has not yet been explained in previous chapters.

### ISSUE THREAD

**Issue by ekalyvio** [progit/progit2#804]
In page, https://git-scm.com/book/en/v2/Git-Basics-Tagging at the end at the paragraph _Checking out Tags_, branching has not yet been explained (what it is) and as such if reading the book sequentially from top to bottom, no meaning is being made of what the author wants to say.

Also, somehow some questions have to be answered. eg:

- are tags independent of which branch you are using? (They don't have any relation with branches? - are the tags same in all branches?)
- what is happening if I push a commit to a remote server that contains a **new** tag but I don't issue the command _git push origin [tagname]_? What is the server going to see?

**Comment by ben** [comment]
Yeah, there are references to branches sprinkled throughout Chapter 2. It's kind of unavoidable, since `git status` mentions them, but in most places we say "don't worry too much about branches for now, we'll cover them in Chapter 3." It looks like this got overlooked.

Do you want to submit a PR to fix the wording up a bit? I'd love for you to get `git blame` credit for making the book better.

**Comment by ekalyvio** [comment]
You mean to do a Pull Request? I am not sure what I should do. I am really newbie in Git and I wouldn't like to mess around on your project unless you quickly instruct me with the steps to do it.

I think that the best way to fix that chapter is:
1) Totally remove the last paragraph _'Checking out Tags'_ from chapter 2.6.
2) Introduce a new quick chapter _'2.7 Checking out commits'_
and 3) shift chapters _'2.7 Git Aliases'_ and _'2.8 Summary'_ to new chapters 2.8 and 2.9 respectively.

In the new chapter _'2.7 Checking out commits'_, I would:
i) Explain what is the HEAD and why in previous chapters we used `git reset HEAD <file>...` to unstage a file (chapter 2.4)
ii) Quickly explain what the `git checkout` command is for (in reality the HEAD is being moved to an other commit).

Also, since the _Commit History_ has been taught (paragraph 2.3), in an example of 3 commits eg. A <- B <- C <- HEAD
I would explain:

iii) How to check out to a previous commit (eg. commit B) by using the SHA-1 checksum (eg. `git checkout 2ff2a0c`) or a Tag (`git checkout 2.0.0`) - part of the _'Checking out Tags'_ paragraph
Here I would show the new log and I would explain that the HEAD has been moved like bellow.

```
* c47dcbb (master) Added 3.txt
* d287397 Added 2.txt
* 2ff2a0c (HEAD) changed line of 1.txt
* 9a0825b Initial commit
```

Then I would instruct the user to go back to the last commit (the master one) if s/he doesn't want to loose any kind of file.

The `git checkout -b version2 v2.0.0` paragraph should move to chapters 3 (_Branching_) and later if it hasn't been explained there.


Finally, in chapter 3 or a later one, an example of a detached head should be introduced by giving an example of a graph by using the command `git log --oneline --decorate --graph --all`:

```
* 7ee76a4 (HEAD) added 2.txt
| * 2ff2a0c (master) changed line of 1.txt
|/
* 9a0825b Initial commit
```

It should be explained that if someone does a `git checkout master` at that point, the commits that do not belong to a branch (like 7ee76a4) will go poof... hidden. 

I haven't read past chapter 3 of the book and as such I am not sure if you somewhere explain the above examples. I just hope I was helpful. :)
